.github/workflows/cd-analytics-infra.yml

```
@@ -52,7 +52,7 @@ jobs:
       fail-fast: false
       matrix:
         directory: ["database", "service"]
-        envs: ${{ inputs.environment || github.event_name == 'release' && fromJSON('["prod"]') || github.ref_name == 'main' && fromJSON('["dev", "staging"]') || fromJSON('["dev"]') }}
+        envs: ${{ fromJSON(inputs.environment != null && format('["{0}"]', inputs.environment) || github.event_name == 'release' && '["prod"]' || github.ref_name == 'main' && '["dev", "staging"]' || '["dev"]')  }}
```

Did you try

```suggestion
        envs: ${{ fromJSON(inputs.environment != null && '["inputs.environment"]' || github.event_name == 'release' && '["prod"]' || github.ref_name == 'main' && '["dev", "staging"]' || '["dev"]')  }}
```

?

Mostly I'm trying to figure out what the `format('["{0}"]'` is in there for. Like I get what it's doing it, I just don't know why we need it